modules/weko-index-tree/weko_index_tree/utils.py

```diff
@@ -596,3 +598,60 @@ def recorrect_private_items_count(agp):
         for bk in bkt:
             if bk.get("from"):
                 agg["no_available"]["doc_count"] += bk.get("doc_count")
+
+
+def check_doi_in_index(index_id):
+    """Check doi in index.
+
+    @param index_id:
+    @return:
+    """
+    from .api import Indexes
+    child_idx = Indexes.get_child_list_by_index(index_id)
+    for index in child_idx:
+        if check_doi_in_list_record(get_record_of_index(index.cid)):
+            return True
+    return False
+
+
+def get_record_of_index(index_id):
+    """Get record of index.
+
+    @param index_id:
+    @return:
+    """
+    from .api import Indexes
+    index_child = Indexes.get_child_list_by_pip(index_id)
+    query = PersistentIdentifier.query \
+        .join(RecordMetadata,
+              PersistentIdentifier.object_uuid == RecordMetadata.id) \
+        .filter(text("CAST(json->>'path' AS text) like '" +
+                     '%"' + index_child[0].path + '"%' + "'")) \
+        .filter(RecordMetadata.id == PersistentIdentifier.object_uuid)
+    list_record_query = query.all()
+    return list_record_query
+
+
+def check_doi_in_list_record(list_record_query):
+    """Check doi in list record.
+
+    @param list_record_query:
+    @return:
+    """
+    list_uuid = dict()
+    for record in list_record_query:
+        if record.object_uuid in list_uuid and int(record.id) > list_uuid.get(
+                record.object_uuid).get('id'):
+            list_uuid[record.object_uuid]['id'] = int(record.id)
+            list_uuid[record.object_uuid]['type'] = record.pid_type
+        else:
+            list_uuid.update(
+                {
+                    record.object_uuid: {
+                        'id': int(record.id),
+                        'type': record.pid_type}
+                })
+    for _, value in list_uuid.items():
+        if value.get('type') == 'doi':
+            return True
+    return False
```

Comment:
About `check_doi_in_index`, you shouldn't loop through chilld indexes, it will cost low perfomance when the amount of indexes is large(ex: Tokyo University is more than 8000 indexes), instead you simple make a query to ES to get all records belonged to that index, then base on these record uuid, you query DB to check DOI. 

Query ES sample:
```
{
    "_source": {
        "excludes": ["content"]
    },
    "query": {
        "bool": {
            "must": [
                {
                    "match": {
                        "path.tree": "***current_index_id***"
                    }
                },
                {
                    "match": {
                        "relation_version_is_last": "true"
                    }
                }
            ]
        }
    }
}
```